app/services/submission/trait_score_parser.rb

```diff
@@ -34,16 +34,20 @@ def call
 
   def map_headers_to_traits
     @upload.log "Mapping file header columns to Trait Descriptors"
-    header = input.readline
+    header = csv.readline
     @trait_names = PlantTrialSubmissionDecorator.decorate(@upload.submission).sorted_trait_names
     @trait_mapping = {}
-    header.split("\t")[1..-1].each_with_index do |column_name, i|
-      next if i >= @trait_names.length
-      column_name = column_name.strip
-      trait_index = @trait_names.find_index(column_name)
-      trait_index = i unless trait_index.present?
-      @trait_mapping[i] = trait_index
-      @upload.log " - Mapped column '#{column_name}' to Trait index #{trait_index} of value #{@trait_names[trait_index]}"
+    if header.blank?
```

Comment:
OK, I think it's fine if you disable it.

Maybe it should be required to start with 'Plant scoring unit name'? It would help to recognize if there is a header line at all. Right now if you skip header altogether the data row (the first row) will be interpreted as header. Of course, that is user's fault, but perhaps we could handle that in a better way.